docs/architecture.md

```diff
@@ -0,0 +1,212 @@
+  - [Controllers](#controllers)
+  - [Cluster Config Controller](#cluster-config-controller)
+  - [Operator Config Controller (Network Controller)](#operator-config-controller-network-controller)
+    - [Applied configuration](#applied-configuration)
+  - [Egress Router](#egress-router)
+  - [Ingress Config](#ingress-config)
+  - [Operator PKI](#operator-pki)
+  - [Signer controller](#signer-controller)
+  - [Proxy Config](#proxy-config)
+  - [Configmap CA Injector](#configmap-ca-injector)
+  - [Connectivity Check Controller](#connectivity-check-controller)
+- [Deriving status](#deriving-status)
+    - [Changes needed](#changes-needed)
+
+This document is an overview of CNO's architecture. It is not an authoritative, detailed reference.
+
+The CNO is collection of _controllers_ that, collectively, configure the network of an OpenShift cluster.
+
+Each controller watches a single type of Kubernetes object, called a Kind or GVK (for Group + Version + Kind). It then creates, updates, or deletes some "downstream" objects as appropriate. Most of the interesting logic is in the individual controllers, which are described below.
+
+[Controllers](https://kubernetes.io/docs/concepts/architecture/controller/) are a central concept in Kubernetes, and CNO follows that logical model.
+
+
+```
+     ┌─────────────────┐
+     │  ┌──────────┐   │
+     │  │          │   │
+     │  │ MyKind   │   │
+     │  │          │   │
+     │  └──────────┘   │
+     │   APIServer     │
+     └─┬───────────────┘
+       │             ▲
+ Watch MyKind.Spec   │
+       │      Update .Status
+       ▼             │
+    ┌────────────────┴──┐
+    │                   │
+    │    Controller     │
+    │                   │
+    └─┬─────────────────┘
+      │              ▲
+Create / Update      │
+      │        Watch .Status
+      ▼              │
+    ┌────────────────┴──┐
+    │                   │
+    │  "Child" objects  │
+    │                   │
+    └───────────────────┘
+```
+
+
+## CNO as SLO
+
+CNO is a so-called second-level-operator (SLO), which means it is installed by the
+Cluster Version Operator (CVO). The CVO has 
+[some documentation](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md)
+about how to interact with it. Most importantly, the CVO will create everything 
+in CNO's `/manifests` folder.
+
+All SLOs have to present a unified API: they watch configuration in the `config.openshift.io` API group, and must report their status in a special object called a [ClusterOperator](https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go).
+
+### Install & Upgrade Orderings
+
+The CVO has a notion of 
+[run levels](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#how-do-i-get-added-as-a-special-run-level),
+which dictate the order in which components are **upgraded**. Presently, the CNO 
+(and thus its operands) are runlevel 07, which is comparatively early. At 
```

Comment:
Perhaps worth clarifying that MCO updates very late, and thus during an upgrade the new networking components will initially be running against an N-1 RHCOS and in particular an N-1 OVS.

Reply:
added something for here.